pyomo/opt/results/solver.py

```diff
@@ -139,42 +139,64 @@ def to_solver_status(tc):
 def check_optimal_termination(results):
     """
     This function returns True if the termination condition for the solver
-    is 'optimal', 'locallyOptimal', or 'globallyOptimal', and the status is 'ok'
+    is 'optimal'. This is backwards and forwards compatible.
 
     Parameters
     ----------
-    results : Pyomo results object returned from solver.solve
+    results : Pyomo Results object returned from solver.solve
 
     Returns
     -------
     `bool`
     """
-    if results.solver.status == SolverStatus.ok and (
-        results.solver.termination_condition == TerminationCondition.optimal
-        or results.solver.termination_condition == TerminationCondition.locallyOptimal
-        or results.solver.termination_condition == TerminationCondition.globallyOptimal
-    ):
-        return True
+    from pyomo.contrib.solver.common.results import (
+        TerminationCondition as FutureTerminationCondition,
+        SolutionStatus as FutureSolutionStatus,
+    )
+
+    if hasattr(results, 'solution_status'):
+        if results.solution_status == FutureSolutionStatus.optimal and (
+            results.termination_condition
+            == FutureTerminationCondition.convergenceCriteriaSatisfied
+        ):
```

Comment:
I don't think there is a need to check the termination condition anymore. Maybe worth a discussion.

Reply:
That's a good point. Let's talk about it in the dev call.